### PR TITLE
[SYCLomatic] Enable cmake migration framework to auto load cmake build-in migration rule file and remove dependency of cuda header files if --migrate-cmake-script-only is specified

### PR DESCRIPTION
**Pull request by tomflinda**
Signed-off-by: chenwei.sun <chenwei.sun@intel.com>

### CODE REVIEW

**Review by zhiweij1 — APPROVED** [review]

File: clang/lib/DPCT/DPCT.cpp [hunk 1]
```diff
@@ -1022,6 +1026,19 @@ int runDPCT(int argc, const char **argv) {
   TypeLocRewriterFactoryBase::initTypeLocRewriterMap();
   MemberExprRewriterFactoryBase::initMemberExprRewriterMap();
   clang::dpct::initHeaderSpellings();
+
+  if (MigrateCmakeScriptOnly || MigrateCmakeScriptOnly) {
```
Comment (zhimingwang36) [inline clang/lib/DPCT/DPCT.cpp]: need update the if statement.

File: clang/lib/DPCT/DPCT.cpp [hunk 2]
```diff
@@ -271,6 +272,16 @@ UnifiedPath getInstallPath(const char *invokeCommand) {
...
 // To validate the root path of the project to be migrated.
 void ValidateInputDirectory(UnifiedPath InRoot) {
+  if (isChildOrSamePath(InRoot, DpctInstallPath)) {
+    ShowStatus(MigrationErrorInputDirContainCTTool);
+    dpctExit(MigrationErrorInputDirContainCTTool);
+  }
+
+  // To skip to check the following cuda path when option
+  // "--migrate-cmake-script-only" is specified to migrate cmake script
+  if (MigrateCmakeScriptOnly)
```
Comment (zhimingwang36) [inline clang/lib/DPCT/DPCT.cpp]: Please don't mix the flag with ValidateInputDirectory().

File: clang/lib/DPCT/DPCT.cpp [hunk 3]
```diff
@@ -805,9 +811,12 @@ int runDPCT(int argc, const char **argv) {
 
   ExtraIncPaths = OptParser->getExtraIncPathList();
 
-  // TODO: implement one of this for each source language.
-  CudaPath = getCudaInstallPath(OriginalArgc, argv);
-  DpctDiags() << "Cuda Include Path found: " << CudaPath.getCanonicalPath() << "\n";
+  if (!MigrateCmakeScriptOnly) {
```
Comment (zhimingwang36) [inline clang/lib/DPCT/DPCT.cpp]: for readability, will it be better if abstract a function like isCUDAHeaderRequired() {return !MigrateCmakeScriptOnly;}?

Comment (tomflinda) [reply clang/lib/DPCT/DPCT.cpp]: Okay

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**